pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go

```diff
@@ -149,6 +151,15 @@ func alterTableAddColumn(
 				typeName.String()))
 		}
 	}
+	// Block unique indexes on unsupported types.
+	if d.Unique.IsUnique &&
+		!d.Unique.WithoutIndex &&
+		!colinfo.ColumnTypeIsIndexable(spec.colType.Type) {
+		msg := "column %s is of type %s and thus is not indexable"
+		typInfo := spec.colType.Type.DebugString()
+		msg = fmt.Sprintf(msg, d.Name, spec.colType.Type.Name())
+		panic(unimplemented.NewWithIssueDetailf(35730, typInfo, "%s", msg))
```

Comment:
nit: if possible, replace this with 
```
panic(unimplemented.NewWithIssueDetailf(35730, typInfo, msg, d.Name, spec.colType.Type.Name())
```
so that the error message redactor doesn't redact the whole message into meaninglessness.